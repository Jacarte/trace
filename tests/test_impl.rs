#[macro_use]
mod trace_test;

use trace::trace;

trace::init_depth_var!();

trace_test!(test_impl, {
    let foo = Foo;
    Foo::foo(2);
    foo.bar(7);
});

struct Foo;

#[trace]
impl Foo {
    fn foo(b: i32) -> i32 {
        b
    }

    fn bar(&self, a: i32) -> i32 {
        a
    }
}
