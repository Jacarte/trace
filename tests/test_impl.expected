[+] Entering foo(b = 2)
[-] Exiting foo = 2
[+] Entering bar(a = 7)
[-] Exiting bar = 7
